File: dl_h2o.r
# Start up H2O
library(h2o)
localH2O = h2o.init()

# Load the dataset
prostate.hex = h2o.uploadFile(localH2O, path = system.file("extdata", "datasets/prostate.csv", package="h2o"), destination_frame = "prostate.hex")
summary(prostate.hex)

# Set the CAPSULE column to be a factor column then build the model
prostate.hex$CAPSULE = as.factor(prostate.hex$CAPSULE)
model = h2o.deeplearning(x = setdiff(colnames(prostate.hex), 
                         c("ID","CAPSULE")), 
                         y = "CAPSULE", 
                         training_frame = prostate.hex, 
                         activation = "Tanh", 
                         hidden = c(10, 10, 10), 
                         epochs = 10000)
print(model@model$model_summary)

# Make predictions with the trained model
predictions = predict(object = model, newdata = prostate.hex)
# Export predictions from H2O Cluster as R dataframe
predictions.R = as.data.frame(predictions)
head(predictions.R)
tail(predictions.R)

# Check performance of the classification model
performance = h2o.performance(model = model)
print(performance)

File: dl_h2o.py
# Fix issue with python install location
import sys
sys.prefix = "/usr/local"

# Start up H2O
import h2o
h2o.init(start_h2o=True)

# Load the dataset
prostate = h2o.upload_file(path=h2o.locate("datasets/prostate.csv"))
prostate.describe()

# Set the CAPSULE column to be a factor column then build the model
prostate["CAPSULE"] = prostate["CAPSULE"].asfactor()
model = h2o.deeplearning(x=prostate[list(set(prostate.col_names()) - set(["ID", "CAPSULE"]))],
                         y = prostate["CAPSULE"],
                         training_frame=prostate,
                         activation="Tanh",
                         hidden=[10, 10, 10],
                         epochs=10000)
model.show()

# Make predictions with the trained model
predictions = model.predict(prostate)
predictions.show()

# Check performance of the classification model
performance = model.model_performance(prostate)
performance.show()

# Domino Diagnostic Statistics
r2 = performance.r2()
mse = performance.mse()
auc = performance.auc()
accuracy = performance.metric('accuracy')[0][1]

import json
with open('dominostats.json', 'wb') as f:
    f.write(json.dumps({"R^2": r2, "MSE": mse, "AUC": auc, "Accuracy": accuracy}))